### ISSUE TITLE
[bug][sdk] When using dsl.OneOf, if the output is used by multiple downstream components the compilation fails

### ISSUE THREAD

**Issue by juansebashr** [kubeflow/pipelines#10271]
### Environment

*  KFP version:
SDK 2.4.0

### Steps to reproduce

I have the following code:
```python
       with dsl.If(with_ingestion == True):
            # Step 0 - Ingestion pipeline
            ingestion_op = ingestion(
               ...
            )
       with dsl.Else():
            dataset_split_op = dataset_split_images(
                ...
            )

            start_stop_instance_op = vm_voxel_components.start_stop_instance(
                ...
            )

       splitted_dataset = dsl.OneOf(ingestion_op.output, dataset_split_op.output)

       training_op = training(
               split_dataset=splitted_dataset
                ...
        )

       export_model_op = export_model(
            ...
        )

       model_ingestion_op = model_ingestion(
            ...
        )
        # old_model_testing_op = test_model_uri(
        #      split_dataset=splitted_dataset
        #    ...
        # )
```

When I uncomment the last component, the compilation fail with this error:

``` 
File "D:\Documents\Kiwibot\ouroboros\pipelines\training\pipeline.py", line 106, in compile_pipeline
    def ouroboros_training(

  File "D:\Documents\Kiwibot\ouroboros\.venv\lib\site-packages\kfp\dsl\pipeline_context.py", line 65, in pipeline
    return component_factory.create_graph_component_from_func(

  File "D:\Documents\Kiwibot\ouroboros\.venv\lib\site-packages\kfp\dsl\component_factory.py", line 669, in create_graph_component_from_func
    return graph_component.GraphComponent(

  File "D:\Documents\Kiwibot\ouroboros\.venv\lib\site-packages\kfp\dsl\graph_component.py", line 68, in __init__
    pipeline_spec, platform_spec = builder.create_pipeline_spec(

  File "D:\Documents\Kiwibot\ouroboros\.venv\lib\site-packages\kfp\compiler\pipeline_spec_builder.py", line 1861, in create_pipeline_spec


  File "D:\Documents\Kiwibot\ouroboros\.venv\lib\site-packages\kfp\compiler\compiler_utils.py", line 524, in get_outputs_for_all_groups
    producer_task = pipeline.tasks[inner_channel.task_name]

KeyError: 'condition-2'
```

### Expected result

The compilation should work without problem

Impacted by this bug? Give it a 👍.

**Comment by xor-xor** [comment]
Any chances of fixing this bug in the nearby future? It is still present in 2.6.0, and I'd say that consuming the output of some task by >1 downstream tasks is nothing exotic/unusal, really (at least in my world).

**Comment by juansebashr** [comment]
For @xor-xor and all people interested on this issue, is solved in 2.7.0

https://github.com/kubeflow/pipelines/pull/10452
